### ISSUE TITLE
Unable to Boot up ElasticSearch

### ISSUE THREAD

**Issue by tiger11** [elastic/elasticsearch#89844]
### Elasticsearch Version

8.5.0-SNAPSHOT

### Java Version

18

### OS Version

x86_64

### Problem Description

I am unable to start Elastic Search on my Linux Machine. When I run ./elasticsearch command, the process exits within 20 seconds with below error.

$ ./elasticsearch -v
Skipping security auto configuration because it appears that security is already configured.

ERROR: Elasticsearch exited unexpectedly
Also there are no relevant logs in logs folder, just GC logs which don't provide any hint. Even -v verbose option is not giving any pointers as to why the service is unable to start.

### Steps to Reproduce

Execute below command
./elasticsearch

**Comment by fcofdez** [comment]
@tiger11 could you include the process exit code in the issue description? thanks

**Comment by tiger11** [comment]
@fcofdez 
I don't see any exit code in console or logs folder.
Below is all that gets printed in console once I execute ./elasticsearch command.

```
$ ./elasticsearch -v
Skipping security auto configuration because it appears that security is already configured.

ERROR: Elasticsearch exited unexpectedly
```

Also in logs folder I dont see any logs except GC logs.

**Comment by fcofdez** [comment]
You can inspect the process exit code with `echo $?`

**Comment by tiger11** [comment]
@fcofdez 

Please check below. Exit code is 137.

```
[elasticsearch@d8364e67a079 bin]$ ./elasticsearch -v
Skipping security auto configuration because it appears that security is already configured.

ERROR: Elasticsearch exited unexpectedly
[elasticsearch@d8364e67a079 bin]$ echo $?
137
[elasticsearch@d8364e67a079 bin]$

```

**Comment by fcofdez** [comment]
It looks like the process runs out of memory.

**Comment by grcevski** [comment]
Elasticsearch will auto configure the heap size based on the available memory on the system, depending on various factors we'll pick something around 40% of the available system memory. You can adjust the heap size manually by modifying the jvm.options file in the config directory.

**Comment by tiger11** [comment]
thanks @fcofdez @grcevski 
I am able to resolve the issue.

**Comment by adarsh4997** [comment]
ERROR: Skipping security auto configuration because it appears that security is already configured.

I'm still getting this in windows while connecting node2 with node1

=>I tried reconfiguring the node2, again got the error:
   ERROR: Aborting enrolling to cluster. Unable to remove existing secure settings. Error was: Aborting enrolling to cluster. Unable to remove existing security configuration, elasticsearch.keystore did not contain expected setting [autoconfiguration.password_hash].
   
to solve this I've added:
C:\Users\ADARSH KUMAR\Downloads\elastic-stack\node-2\bin>elasticsearch-keystore add autoconfiguration.password_hash

Now when I'm again reconfiguring I'm getting this:
ERROR: Skipping security auto configuration because this node is configured to bootstrap or to join a multi-node cluster, which is not supported.


Can someone help?
